src/screens/home-staking.tsx

```diff
@@ -1,31 +1,79 @@
 import React, {useCallback, useEffect, useRef, useState} from 'react';
 
 import {HomeStaking} from '@app/components/home-staking';
-import {Loading} from '@app/components/ui';
-import {app} from '@app/contexts';
-import {useTypedNavigation, useWallets} from '@app/hooks';
+import {useApp, useTypedNavigation, useWalletsList} from '@app/hooks';
+import {I18N, getText} from '@app/i18n';
+import {StakingMetadata} from '@app/models/staking-metadata';
 import {Cosmos} from '@app/services/cosmos';
 
+const initData = {
+  stakingSum: 0,
+  rewardsSum: 0,
+  unDelegationSum: 0,
+  loading: true,
+};
+
 export const HomeStakingScreen = () => {
-  const [loading, setLoading] = useState(true);
-  const cosmos = useRef(new Cosmos(app.provider!)).current;
-  const wallets = useWallets();
+  const app = useApp();
+  const [data, setData] = useState(initData);
+
+  const {visible} = useWalletsList();
   const navigation = useTypedNavigation();
+  const cosmos = useRef(new Cosmos(app.provider!)).current;
+
   const onPressValidators = useCallback(() => {
     navigation.navigate('stakingValidators');
   }, [navigation]);
 
   useEffect(() => {
-    const addressList = wallets.visible.map(w => w.cosmosAddress);
+    const newData = StakingMetadata.getSummaryInfo();
+    setData({...newData, loading: false});
+  }, []);
+
```

Comment:
add listener like StakingMetadata.getAll().addListener() and on change recalculate summary

Reply:
or recalculate summary after `cosmos.sync(addressList)` finished